src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,482 @@
+  geometries = Union{Geometry,Missing,Nothing}[]
+  
+  # Use Tables.rows interface
+  for row in Tables.rows(geom_result)
+    # Use Tables.getcolumn for accessing column values
+    blob = Tables.getcolumn(row, Symbol(geom_column))
+    if ismissing(blob)
+      push!(geometries, missing)
+    elseif isnothing(blob)
+      push!(geometries, nothing)
+    else
+      try
+        geom = parse_gpb(blob, crs_type)
+        push!(geometries, geom)
+      catch
+        # If parsing fails, treat as missing geometry
+        push!(geometries, missing)
+      end
+    end
+  end
+  
+  return geometries
+end
+
+# Get CRS from SRID according to GeoPackage specification
+# Reference: https://www.geopackage.org/spec/#gpkg_spatial_ref_sys_cols
+# SRID possibilities:
+#  0: Undefined geographic coordinate reference system
+# -1: Undefined Cartesian coordinate reference system  
+#  1-32766: Reserved for OGC use (EPSG codes)
+#  32767-65535: Reserved for GeoPackage use
+#  >65535: User-defined coordinate reference systems
+function get_crs_from_srid(db::SQLite.DB, srid::Integer)
+  # Handle special cases per GeoPackage spec
+  if srid == 0
+    # Undefined geographic coordinate reference system - use WGS84 as default
+    return LatLon{WGS84Latest}
+  elseif srid == -1
+    # Undefined Cartesian coordinate reference system
+    return Cartesian{NoDatum}
+  end
+  
+  # Query the database for the organization and organization_coordsys_id
+  srs_result = DBInterface.execute(db,
+    """
+    SELECT organization, organization_coordsys_id
+    FROM gpkg_spatial_ref_sys
+    WHERE srs_id = ?
+    """, [srid])
+  
+  # Try to get the organization info
+  org_info = nothing
+  for row in srs_result
+    org_info = row
+    break
+  end
+  
+  # According to GeoPackage spec, this should not happen in compliant files
+  # but handle gracefully for malformed GeoPackages
+  if isnothing(org_info)
+    @warn "SRID $srid not found in gpkg_spatial_ref_sys table. This violates GeoPackage specification."
+    return Cartesian{NoDatum}
+  end
+  
+  # Extract organization name - spec requires organization to be NOT NULL
+  # but handle malformed files gracefully
+  has_org_name = !isnothing(org_info.organization) && !ismissing(org_info.organization)
+  if !has_org_name
+    @warn "Missing organization for SRID $srid. This violates GeoPackage specification (organization is NOT NULL)."
+  end
+  org_name = has_org_name ? uppercase(org_info.organization) : ""
+  
+  # Extract coordinate system ID - spec requires organization_coordsys_id to be NOT NULL
+  # but handle malformed files gracefully
+  has_coord_id = !isnothing(org_info.organization_coordsys_id) && !ismissing(org_info.organization_coordsys_id)
+  if !has_coord_id
+    @warn "Missing organization_coordsys_id for SRID $srid. This violates GeoPackage specification (organization_coordsys_id is NOT NULL)."
+    return Cartesian{NoDatum}
+  end
+  coord_id = org_info.organization_coordsys_id
```

Comment:
I confess that I am not feeling that you are reading the spec carefully. Every time I comment on the PR, you adjust things without providing sources. Please share the link to the spec section that discusses all these details.